### DISCUSSION TITLE
keep a branch updated

### DISCUSSION THREAD

**Discussion by aster94** [community/community#22307] [New to GitHub]
<p>hello,</p>
<p>I have a repository where the main code is in the master but i have a few branches with minimal features which i didn’t implemented yet in the master branch. How could I maintain these branch updated with the newer code inside master?</p>

**Comment by Wabri** [comment, marked as answer]
<p>Hi <a class="mention" href="/aster94">@aster94</a>,</p>
<p>You can merge the master into the branch or cherry-pick the commits that you want in these branches.</p>
<p><strong>Merge solution</strong></p>
<p>From github web interface you can click in the “new pull request” button and choose to merge master into one of your active branch.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/9/9cfc89b69f3fe36719283e7c094759b1ab53ce25.png" data-download-href="/uploads/short-url/moLxOEY2VL5iJwF5hUqUz73GbpH.png?dl=1" title="latest-screenshot.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/12409541/181070134-ed5bd31f-6466-43ca-aa1d-a0e926636421.png" alt="latest-screenshot.png" data-base62-sha1="moLxOEY2VL5iJwF5hUqUz73GbpH" width="690" height="72" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/9/9cfc89b69f3fe36719283e7c094759b1ab53ce25_2_690x72.png, https://aws1.discourse-cdn.com/github/original/2X/9/9cfc89b69f3fe36719283e7c094759b1ab53ce25.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/9/9cfc89b69f3fe36719283e7c094759b1ab53ce25.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/9/9cfc89b69f3fe36719283e7c094759b1ab53ce25_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">latest-screenshot.png</span><span class="informations">989×104 17.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>When you merge the master into the other branch remember that all the changes made on master will be made even on that branch.</p>
<p><strong>Cherry-pick solution</strong></p>
<p><em>You can find all the documentation of cherry-pick here: <a href="https://git-scm.com/docs/git-cherry-pick" rel="nofollow noopener">git-cherry-pick</a></em></p>
<p>Get the <strong>commit ID</strong> of the commit you want on the branch.</p>
<p>You can get the commit id by the web UI:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/1/1f3c49fd8278e20957d190611df85c4c9ff97e04.png" data-download-href="/uploads/short-url/4sjY9xlTMwDbv231rbCkCKVtKg4.png?dl=1" title="aaaaaaaaa.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/12409541/181070136-072b03f0-1ba4-4c99-9d53-d607723eff9d.png" alt="aaaaaaaaa.png" data-base62-sha1="4sjY9xlTMwDbv231rbCkCKVtKg4" width="690" height="78" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/1/1f3c49fd8278e20957d190611df85c4c9ff97e04_2_690x78.png, https://aws1.discourse-cdn.com/github/optimized/2X/1/1f3c49fd8278e20957d190611df85c4c9ff97e04_2_1035x117.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/1/1f3c49fd8278e20957d190611df85c4c9ff97e04_2_1380x156.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/1/1f3c49fd8278e20957d190611df85c4c9ff97e04_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">aaaaaaaaa.png</span><span class="informations">1500×171 29.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/7/773391699a95a5a6d6360f9e21797fa94427349b.png" data-download-href="/uploads/short-url/h0viMxRMk2dDtoxGt0vRbn3bNvt.png?dl=1" title="bbbbbbbbbbb.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/12409541/181070138-89b82b1f-ffdb-42bb-964f-3a9927539ee2.png" alt="bbbbbbbbbbb.png" data-base62-sha1="h0viMxRMk2dDtoxGt0vRbn3bNvt" width="690" height="159" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/7/773391699a95a5a6d6360f9e21797fa94427349b_2_690x159.png, https://aws1.discourse-cdn.com/github/optimized/2X/7/773391699a95a5a6d6360f9e21797fa94427349b_2_1035x238.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/7/773391699a95a5a6d6360f9e21797fa94427349b_2_1380x318.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/7/773391699a95a5a6d6360f9e21797fa94427349b_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">bbbbbbbbbbb.png</span><span class="informations">1511×350 67.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Or with command line:</p>
<pre><code>git log
</code></pre>
<p>with output:</p>
<pre><code>commit&lt;u&gt;&lt;strong&gt;0624810957941c930b8314d15b1fa9538c54b98c&lt;/strong&gt;&lt;/u&gt; **\&lt;--------------------- this is the commit id** Author: Gabriele Puliti &lt;gabriele.puliti@gmail.com&gt;
Date: Sun Feb 24 19:18:51 2019 +0100

    update (config): create some fast shortcut
</code></pre>
<p>Copy this value and run the cherry-pick command with the commit ID you have copied (you need to be in the branch where you want this changes):</p>
<pre><code>git cherry-pick&lt;u&gt;&lt;strong&gt;0624810957941c930b8314d15b1fa9538c54b98c&lt;/strong&gt;&lt;/u&gt;
</code></pre>
<p>And this is it.</p>
<p>Hope this helps :slightly_smiling_face:  </p>
<p>-Gabriele-</p>
<blockquote>
<p><strong>Mark helpful posts with Accept as Solution to help other users locate important info. Don’t forget to give Kudos for great content!</strong></p>
</blockquote>

**Comment by aster94** [comment]
<p>The merge solution is problably what would fit better my needs, but I am also tto lazy to merge every time master inside the branches :grin:</p>
<p>When i would be ready to merge the new features inside master i would use the cherry-pick :+1:</p>
<p>grazie Gabriele!</p>
